### PR TITLE
conditionally enable allow-egress LBs

### PR DESCRIPTION
**Pull request by kon-angelo**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform azure

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
Conditionally enable allow-egress-{tcp,udp} loadbalancers.
```

### CODE REVIEW

File: pkg/controller/infrastructure/infraflow/ensurer.go [hunk 1]
```diff
@@ -710,14 +710,24 @@ func (fctx *FlowContext) GetInfrastructureStatus(_ context.Context) (*v1alpha1.I
...
-		})
+		}
+		if z.NatGateway != nil {
+			subnet.NatGatewayName = ptr.To(z.NatGateway.Name)
+		}
+		status.Networks.Subnets = append(status.Networks.Subnets, subnet)
+		// if at least one of the "zones" does not have NATGateway enabled, mark the outbound access as OutboundAccessTypeLoadBalancer.
+		if z.NatGateway == nil {
+			outboundAccessType = v1alpha1.OutboundAccessTypeLoadBalancer
+		}
```
Comment (AndreasBurger) [inline pkg/controller/infrastructure/infraflow/ensurer.go:728]:
```suggestion
		if z.NatGateway != nil {
			subnet.NatGatewayName = ptr.To(z.NatGateway.Name)
		} else {
		         // if at least one of the "zones" does not have NATGateway enabled, mark the outbound access as OutboundAccessTypeLoadBalancer.
		        outboundAccessType = v1alpha1.OutboundAccessTypeLoadBalancer
		}
		status.Networks.Subnets = append(status.Networks.Subnets, subnet)


```

**Review by AndreasBurger — DISMISSED** [review]
lgtm

**Comment by kon-angelo** [comment]
@AndreasBurger PTAL :)

File: pkg/controller/controlplane/valuesprovider.go [hunk 2]
```diff
@@ -727,7 +727,12 @@ func getControlPlaneShootChartValues(
 		"global": map[string]interface{}{
 			"vpaEnabled": gardencorev1beta1helper.ShootWantsVerticalPodAutoscaler(cluster.Shoot),
 		},
-		azure.AllowEgressName: map[string]interface{}{"enabled": infraStatus.Zoned || azureapihelper.IsVmoRequired(infraStatus)},
+		// the allow-egress chart is enable in all cases **except**:
```
Comment (hebelsan) [inline pkg/controller/controlplane/valuesprovider.go:730]:
```suggestion
		// the allow-egress chart is enabled in all cases **except**:
```

**Review by hebelsan — DISMISSED** [review]
/lgtm

**Review by hebelsan — APPROVED** [review]

**Pull request merged**